evm_loader/lib/src/types/mod.rs

```diff
@@ -66,20 +68,38 @@ impl TxParams {
                 .map(|a| (a.address, a.storage_keys))
                 .collect();
 
-            let access_list_tx = AccessListTx {
-                nonce,
-                gas_price: U256::ZERO,
-                gas_limit: self.gas_limit.unwrap_or(U256::MAX),
-                target: self.to,
-                value: self.value.unwrap_or_default(),
-                call_data: self.data.unwrap_or_default(),
-                chain_id: U256::from(chain_id),
-                access_list,
-                r: U256::ZERO,
-                s: U256::ZERO,
-                recovery_id: 0,
-            };
-            TransactionPayload::AccessList(access_list_tx)
+            if let Some(max_priority_fee_per_gas) = self.max_priority_fee_per_gas {
+                let dynamic_fee_tx = DynamicFeeTx {
+                    nonce,
+                    max_priority_fee_per_gas,
+                    max_fee_per_gas: self.max_fee_per_gas.unwrap_or(max_priority_fee_per_gas * 2),
```

Comment:
Just wondering if the 2 here is arbitrary or something agreed on.

Reply:
for library code I don't this it's very relevant (it's not used currently). The only catch is to have base_fee_per_gas > 0 , otherwise some validation may fail when this code is called from the api/cli for example.

base_fee_per_gas = max_fee_per_gas - max_priority_fee_per_gas =>

So I just assigned base_fee_per_gas = max_priority_fee_per_gas and both are valid.